### PR TITLE
GLM: Add version 0.9.9.8.

### PR DESCRIPTION
**Pull request by RemiLacroix-IDRIS**
The CMake install target was removed in version 0.9.9.6 so update the package accordingly.

### CODE REVIEW

File: var/spack/repos/builtin/packages/glm/package.py [hunk 1]
```diff
@@ -12,8 +12,26 @@ class Glm(CMakePackage):
...
 
     homepage = "https://github.com/g-truc/glm"
-    url = "https://github.com/g-truc/glm/archive/0.9.7.1.tar.gz"
+    url = "https://github.com/g-truc/glm/archive/0.9.9.8.tar.gz"
 
+    version('0.9.9.8', sha256='7d508ab72cb5d43227a3711420f06ff99b0a0cb63ee2f93631b162bfe1fe9592')
     version('0.9.7.1', sha256='285a0dc8f762b4e523c8710fbd97accaace0c61f45bc8be2bdb0deed07b0e6f3')
 
-    depends_on('cmake@2.6:', type='build')
+    depends_on('cmake@3.2:', type='build')
```
Comment (becker33) [inline var/spack/repos/builtin/packages/glm/package.py]: If this requirement was accurate for the previous version, we should add a new restriction only for the newer version.

```suggestion
    depends_on('cmake@2.6:', type='build')
    depends_on('cmake@3.2:', type='build', when='@0.9.9.8:')
```

File: var/spack/repos/builtin/packages/glm/package.py [hunk 2]
```diff
@@ -12,8 +12,26 @@ class Glm(CMakePackage):
...
+    def build(self, spec, prefix):
+        pass
+
+    @when('@0.9.9.6:')
+    def install(self, spec, prefix):
+        mkdirp(prefix.include.glm)
+        ignore_cmakelists = lambda p: p.endswith('CMakeLists.txt')
+        install_tree('glm', prefix.include.glm, ignore=ignore_cmakelists)
+        mkdirp(prefix.lib64.cmake)
+        install_tree('cmake', prefix.lib64.cmake)
```
Comment (becker33) [inline var/spack/repos/builtin/packages/glm/package.py:38]: This doesn't look like it depends on cmake at all, it looks like it just copies files over. Why was the cmake constraint added in this PR?

**Review by becker33 — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/glm/package.py (hunk 2 above)
Comment (RemiLacroix-IDRIS) [reply var/spack/repos/builtin/packages/glm/package.py:38]: Let me double-check that, it's indeed weird.

Comment (RemiLacroix-IDRIS) [reply var/spack/repos/builtin/packages/glm/package.py:38]: Ok, the dependency on `cmake@3.2:` was introduced in version 0.9.9.0. I have repushed.

I don't think I can remove the CMake dependency completely as of version 0.9.9.6 (or at least not when inheriting from `CMakePackage`).

**Comment by RemiLacroix-IDRIS** [comment]
@becker33 @tldahlgren : Is there anything preventing from merging this PR? I just realized that it's needed for the VAPOR package I introduced in #22477.

File: var/spack/repos/builtin/packages/glm/package.py [hunk 3]
```diff
@@ -12,8 +12,27 @@ class Glm(CMakePackage):
     """
 
     homepage = "https://github.com/g-truc/glm"
-    url = "https://github.com/g-truc/glm/archive/0.9.7.1.tar.gz"
+    url = "https://github.com/g-truc/glm/archive/0.9.9.8.tar.gz"
 
+    version('0.9.9.8', sha256='7d508ab72cb5d43227a3711420f06ff99b0a0cb63ee2f93631b162bfe1fe9592')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/glm/package.py:17]: Confirmed the sha256 is valid for the new version.

**Comment by tldahlgren** [comment]
> @becker33 @tldahlgren : Is there anything preventing from merging this PR? I just realized that it's needed for the VAPOR package I introduced in #22477.

I will defer to @becker33 on this PR.

**Comment by becker33** [comment]
I'm ok merging it, in the long run we should go back and make it not depend on cmake when it doesn't need to.

**Review by becker33 — APPROVED** [review]

**Comment by RemiLacroix-IDRIS** [comment]
> I'm ok merging it, in the long run we should go back and make it not depend on cmake when it doesn't need to.

Hmm I didn't quite get that it was possible but I'm willing to try.

I thought about having `depends_on('cmake', when='@0.9.9.5:')` but it doesn't feel like it cannot work since `depends_on('cmake')` would be inherited from `CMakePackage`.

Am I correct in saying that not depending on CMake will require getting ride of `CMakePackage` entirely?

**Pull request merged**